wicket-extensions/src/main/java/org/apache/wicket/extensions/ajax/markup/html/AjaxLazyLoadPanel.java

```diff
@@ -187,8 +237,56 @@ public Component getLoadingComponent(final String markupId)
 	{
 		IRequestHandler handler = new ResourceReferenceRequestHandler(
 			AbstractDefaultAjaxBehavior.INDICATOR);
-		return new Label(markupId, "<img alt=\"Loading...\" src=\"" +
-			RequestCycle.get().urlFor(handler) + "\"/>").setEscapeModelStrings(false);
+		return new Label(markupId,
+			"<img alt=\"Loading...\" src=\"" + RequestCycle.get().urlFor(handler) + "\"/>")
+				.setEscapeModelStrings(false);
 	}
 
+	/**
+	 * Factory method for creating the lazily loaded component that replaces the spinner after
+	 * {@link #isReadyForReplacement()} returns {@code true}. You may call setRenderBodyOnly(true)
+	 * on this component if you need the body only.
+	 * 
+	 * @param markupId
+	 *            The components markupid.
+	 * @return The component that must be lazy created.
+	 */
+	public abstract Component getLazyLoadComponent(String markupId);
+
+	/**
+	 * The AJAX timer for updating the AjaxLazyLoadPanel. Is designed to be a page-local singleton
+	 * keeping track of multiple LazyLoadPanels using reference counting.
+	 */
+	private class AjaxLazyLoadTimer extends AbstractAjaxTimerBehavior
```

Comment:
It would not pose a problem and make the code simpler. With static I have to get the component's page and ran into a order problem of removing the behavior from the page and trying to change the page's meta data. The dangling reference is only a problem when you have memory constraints, but then the lazyloadpanel's reference would not be your biggest problem. Regardless I've changed it to be static.